src/psij/executors/batch/pbspro.py

```diff
@@ -0,0 +1,118 @@
+from pathlib import Path
+from typing import Optional, Collection, List, Dict, TextIO
+
+from psij import Job, JobStatus, JobState, SubmitException
+from psij.executors.batch.batch_scheduler_executor import BatchSchedulerExecutor, \
+    BatchSchedulerExecutorConfig, check_status_exit_code
+from psij.executors.batch.script_generator import TemplatedScriptGenerator
+
+import json
+
+_QSTAT_COMMAND = 'qstat'
+
+_STATE_MAP = {
+    'Q': JobState.QUEUED,
+    'R': JobState.ACTIVE,
+
+    'F': JobState.COMPLETED,  # This happens for failed jobs too, so
+                              # need to rely on richer JSON status codes
+                              # and .ec file handling for failure detection
+
+    'E': JobState.ACTIVE      # This state could ambiguously be either the
+                              # very end of being ACTIVE or the very
+                              # beginning of being COMPLETED. This mapping
+                              # treats it as ACTIVE because not all final
+                              # job information has appeared in JSON yet.
+}
+
+
+class PBSProExecutorConfig(BatchSchedulerExecutorConfig):
+    """A configuration class for the PBS executor.
+    This doesn't have any fields in addition to BatchSchedulerExecutorCOnfig,
+    but it is expected that some will appear during further development."""
+
+    pass
+
+
+class PBSProJobExecutor(BatchSchedulerExecutor):
+    """A :proc:`~psij.JobExecutor` for PBS Pro."""
+
+    # TODO: find a comprehensive list of possible states. at least look in parsls state map.
+    def __init__(self, url: Optional[str] = None, config: Optional[PBSProExecutorConfig] = None):
+        """Initializes a :proc:`~PBSProJobExecutor`."""
+        if not config:
+            config = PBSProExecutorConfig()
+        super().__init__(url=url, config=config)
+        self.generator = TemplatedScriptGenerator(config, Path(__file__).parent / 'pbspro'
+                                                  / 'pbspro.mustache')
+
+    # Submit methods
+
+    def generate_submit_script(self, job: Job, context: Dict[str, object],
+                               submit_file: TextIO) -> None:
+        """See :proc:`~BatchSchedulerExecutor.generate_submit_script`."""
```

Comment:
This may be correct but I've never seen ``:proc:`` before... Is that a real directive?

Reply:
Maybe, maybe not - but it isn't used in the rest of the code base so it would be more consistent to make it :meth:

Reply:
I've made that change